yarpcconfig/builder.go

```diff
@@ -326,3 +329,13 @@ func (b *builder) AddStreamOutbound(
 func (b *builder) needTransport(spec *compiledTransportSpec) {
 	b.needTransports[spec.Name] = spec
 }
+
+func tlsModeDecodeHook(from, to reflect.Type, data reflect.Value) (reflect.Value, error) {
+	var mode yarpctls.Mode
+	if from.Kind() != reflect.String || to != reflect.TypeOf(mode) {
```

Comment:
do you want to create a const like what we have in `spec.go` around L247 for this mode?

Reply:
Consts in spec.go are for pointer types, but mode here is a non-pointer type so thought of keeping it here